assets/js/hooks/useUpdateAnimation.tsx

```diff
@@ -0,0 +1,28 @@
+import { useEffect, useState } from "react";
+
+export const useUpdateAnimation = (
+  deps: any[],
+  prevValue: any,
+  showAnimationOnMount?: boolean
+) => {
+  const [showAnimation, setShowAnimation] = useState(false);
+  const [timer, setTimer] = useState<NodeJS.Timeout>();
+  useEffect(() => {
+    // Prevents animation when already showing and on page load.
+    if (timer || (!showAnimationOnMount && !prevValue)) {
+      return;
+    }
+
+    setShowAnimation(true);
+    const newTimer = setTimeout(() => {
+      setTimer(undefined);
+      setShowAnimation(false);
+    }, 2000);
+    setTimer(newTimer);
+
+    // Cancel timer is component unmounts.
+    return () => clearTimeout(newTimer);
```

Comment:
Unsure if this is what we want. Here's a case I'm imagining:

1. `deps` change. -> We start a timer and save its ID in the `timer` state.
2. `deps` change again within 2 seconds. -> React runs the cleanup function immediately before running the effect again, clearing the timer before it triggers. The effect runs and sees that we still have a timer ID stored in the `timer` state, so it does nothing.
3. Now, we have a timer ID stuck in the `timer` state with no logic to ever remove it. Animations on this component stop happening for the remainder of the component's existence on the page, because every time the effect runs, it stops short at `if (timer || ...) {return;}`.

Could you double check whether the hook actually behaves as I described if 2 updates happen within 2 seconds? It's very tricky to visualize timing stuff, so I might be wrong.

If it _does_ happen, I think you will need to have the timer-clearing logic run only when the component unmounts, and not every time it re-renders. The Internet:tm: recommends splitting the logic into two effects. The first effect will be the one you already have, but without a cleanup callback. The second effect will do nothing and have no dependencies—it will simply return the cleanup callback to be run on unmount.

```js
useEffect(() => {
  const cleanup = () => clearTimeout(timer);
  return cleanup;
}, [] );
```

or more succinctly:
```js
useEffect(() => () => clearTimeout(timer), []);
```

> **When exactly does React clean up an effect?** React performs the cleanup when the component unmounts. However, as we learned earlier, effects run for every render and not just once. This is why React _also_ cleans up effects from the previous render before running the effects next time.

Reply:
Would our decision to fetch new alerts every 4 seconds prevent multiple updates in 2 seconds? I think it would because the only `deps` that can trigger an animation come from the same `fetch` that only happens every 4 seconds. The Screenplay client doesn't directly manipulate any of the `deps` (except maybe the places and screens counts, but even those numbers are based on results from the `fetch`).

Reply:
I think it would, but I'm sliiiightly worried about implicitly baking that assumption into the code. If you think the proposed fix would decrease readability or something, I'd be willing to let it go, but if at all possible I'd prefer the hook to work the same regardless of how frequently a component re-renders.

Reply:
Makes total sense, and doesn't decrease readability. Just took me a minute to wrap my head around the problem. I think I have it. Needed to make a small tweak though. In a separate comment, I said that we only need to check for `prevValue !== undefined`, but that isn't exactly true. We need to also make sure it isn't `null`. With our strict typing, those would need to be two separate checks, or we could check for a falsy value.

Reply:
I think that's one of the few times when the less-strict `!=` (or `==`) are recommended—`x != undefined`, or `x != null`, returns true when `x` is either `undefined` or `null`. But no worries, the separate checks are fine.

Reply:
Somehow my brain forgot that you didn't have to use strict equality 😬 . That's a much better idea. Changed to that just in case `prevValue` is a boolean in the future.